### PR TITLE
Async chat ii

### PR DESCRIPTION
**Pull request by Wirt4**
This branch continues the user-side invocation of the chat functionality. The chats are sorted by chat id, and the chat ID page contains a couple of subcomponents: Messages and ChatInput. The ChatId page's functionality itself has also been extended.

The Messsages component displays messages retrieved from the backend and styles them according to whether the sender is the session user or their correspondent.

The ChatInput component uses an api endpoint that hasn't been implemented yet, styled and setup to use an as-yet not implemented API endpoint. It uses 'react-textarea-autosize' plugin.

A function signature has already been updated.

To sum up, this should be everything a user needs to send chat messages short of the API itself.

### CODE REVIEW

File: next.config.mjs [hunk 1]
```diff
@@ -1,4 +1,8 @@
 /** @type {import('next').NextConfig} */
-const nextConfig = {};
+const nextConfig = {
+    images: {
+        domains: ["lh3.googleusercontent.com"],
```
Comment (Wirt4) [inline next.config.mjs:4]: can notes be added here to explain why

File: src/app/(dashboard)/dashboard/chat/[chatId]/page.tsx [hunk 2]
```diff
@@ -10,12 +37,65 @@ interface ChatProps{
...
 const Page: FC<ChatProps> = async ({params}) => {
     const session = await myGetServerSession();
-    if (!session || !params.chatId.split('--').includes(session.user.id)){
+    const userId = session?.user?.id as string
+    const {chatId} = params
+    const participants = new Participants(chatId, userId as string)
+    const helpers = new Helpers()
+    const initialMessages = await helpers.getChatMessages(chatId)
+
+    if (!(session && participants.includesSession())){
```
Comment (Wirt4) [inline src/app/(dashboard)/dashboard/chat/[chatId]/page.tsx:46]: don't need to evaluate session here. If session is null, then User ID will be an empty string.Add a test case for this if it doesn't exist already and see if that check can be removed

File: src/app/(dashboard)/dashboard/chat/[chatId]/page.tsx [hunk 3]
```diff
@@ -10,12 +37,65 @@ interface ChatProps{
...
         notFound();
     }
 
-    return <div/>
+    const partner = (await db.get(`user:${participants.partnerId()}`)) as User;
```
Comment (Wirt4) [inline src/app/(dashboard)/dashboard/chat/[chatId]/page.tsx:50]: abstract the template string into the partners object

File: src/components/ChatInput/ChatInput.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import TextareaAutosize from 'react-textarea-autosize';
+import Helpers from "@/components/ChatInput/helpers";
+import Button from "@/components/ui/button/Button";
+
+interface ChatInputProps {
+    chatPartner: User,
+    chatId: string,
+}
+
+
```
Comment (Wirt4) [inline src/components/ChatInput/ChatInput.tsx:13]: extra space here

File: src/components/ChatInput/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    chatId: string
+    endpoint: string = '/api/message/send'
+
+    constructor( setters: StateSetters, chatId: string) {
+        this.setters = setters;
+        this.chatId = chatId
+    }
+
+    HandleKeystroke  (event: React.KeyboardEvent<HTMLTextAreaElement>, input:string) {
+        if (event.key === "Enter" && !event.shiftKey) {
```
Comment (Wirt4) [inline src/components/ChatInput/helpers.ts:22]: abstract here for self-documentation

File: src/components/Messages.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+const Messages: FC<MessagesProps> = ({initialMessages, sessionId}) => {
+    const [messages, setMessages] = useState<Message[]>(initialMessages)
+    const scrollDownRef = useRef<HTMLDivElement | null>(null)
+
+    return <div aria-label='messages' className='message-scroll'>
+        <div ref={scrollDownRef}>
+        {messages.map((message, index) => {
+            const foo = messages[index - 1]?.senderId === messages[index].senderId
```
Comment (Wirt4) [inline src/components/Messages.tsx:18]: better name for :foo", also  abstract the assignment into a function

File: src/components/Messages.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+class ClassNames {
+    private readonly isCurrentUser: boolean
+    private readonly hasNextMessage: boolean
+
+    constructor(isCurrentUser: boolean, hasNextMessageFromSameUser: boolean) {
+        this.isCurrentUser = isCurrentUser
+        this.hasNextMessage = hasNextMessageFromSameUser
+    }
+
+    class(base: string, ifIsUserId: string, ifNotUserId: string | null = null){
```
Comment (Wirt4) [inline src/components/Messages.tsx:53]: vague signature: try "className"

File: src/components/Messages.tsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        const suffix = this.isCurrentUser ? ifIsUserId : ifNotUserId
+        if (!suffix){
+            return base
+        }
+        return `${base} ${suffix}`
+    }
+
+    get div1(): string{
```
Comment (Wirt4) [inline src/components/Messages.tsx:61]: Move everything here into globals, get anal about the names so it clear to focus on the branching logis
